Tasks/AzureKeyVault/task.json

```diff
@@ -0,0 +1,102 @@
+{
+    "id": "1E244D32-2DD4-4165-96FB-B7441CA9331E",
+    "name": "AzureKeyVault",
+    "friendlyName": "Azure Key Vault",
+    "description": "Download Azure Key Vault Secrets",
+    "helpMarkDown": "[More Information](https://aka.ms/azkvtaskreadme)",
+    "category": "Deploy",
+    "releaseNotes": "Works with Xplat agents (Windows, Linux or OSX)",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 1,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "demands": [],
+    "minimumAgentVersion": "2.0.0",
+    "groups": [
+        {
+            "name": "AzureKeyVaultDetails",
+            "displayName": "Azure key vault details",
+            "isExpanded": true
+        }
+    ],
+    "inputs": [
+        {
+            "name": "ConnectedServiceName",
+            "type": "connectedService:AzureRM",
+            "label": "Azure subscription",
+            "defaultValue": "",
+            "required": true,
+            "groupName": "AzureKeyVaultDetails",
+            "helpMarkDown": "Select the Azure subscription for the key vault."
+        },
+        {
+            "name": "KeyVaultName",
+            "type": "pickList",
+            "label": "Key vault",
+            "required": true,
+            "groupName": "AzureKeyVaultDetails",
+            "helpMarkDown": "Provide the name of a key vault.",
```

Comment:
Is there any allowed set of characters and restrictions for name. Please call that out in help. It might also be good idea to validate that in the script.